From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <blake.galbreath@gmail.com>
Date: Sat, 19 Oct 2019 01:42:50 -0500
Subject: [PATCH] Fix SpawnChangeEvent not firing for all use-cases


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 858493359c04d5eb80285dd0760bea4f247aff8c..5b03dfe6e4c35e59dca5090dc5204b12df33fd04 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1432,8 +1432,11 @@ public abstract class World implements GeneratorAccess, AutoCloseable {
         return blockposition;
     }
 
+    public void setSpawn(BlockPosition blockposition) { v(blockposition); } // Purpur - OBFHELPER
     public void a_(BlockPosition blockposition) {
+        BlockPosition prevPos = getSpawn(); // Purpur
         this.worldData.setSpawn(blockposition);
+        new org.bukkit.event.world.SpawnChangeEvent(world, MCUtil.toLocation(this, prevPos)).callEvent(); // Purpur
     }
 
     public boolean a(EntityHuman entityhuman, BlockPosition blockposition) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 4aafad0762df44990ed9d610a59f3862bd3fe3b0..dcf608fcaa29c2d27702bf0c9d26fb9b2ed23c37 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -383,12 +383,16 @@ public class CraftWorld implements World {
     @Override
     public boolean setSpawnLocation(int x, int y, int z) {
         try {
-            Location previousLocation = getSpawnLocation();
-            world.worldData.setSpawn(new BlockPosition(x, y, z));
+            // Purpur start - move to World#setSpawn()
+            //Location previousLocation = getSpawnLocation();
+            //world.worldData.setSpawn(new BlockPosition(x, y, z));
 
             // Notify anyone who's listening.
-            SpawnChangeEvent event = new SpawnChangeEvent(this, previousLocation);
-            server.getPluginManager().callEvent(event);
+            //SpawnChangeEvent event = new SpawnChangeEvent(this, previousLocation);
+            //server.getPluginManager().callEvent(event);
+
+            world.setSpawn(new BlockPosition(x, y, z));
+            // Purpur end
 
             return true;
         } catch (Exception e) {
