From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 12 Mar 2014 23:10:04 -0400
Subject: [PATCH] EMC LivingEntityArmorProtectEvent

To control if armor should protect entity, how much if so, and how much item damage to give.

diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 741513faf2def6367964ce6cfefe7dd5a87dfc92..b40ba3a21142c0c31f323523d83784e26581eabe 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -39,6 +39,7 @@ import org.bukkit.event.player.PlayerItemConsumeEvent;
 // CraftBukkit end
 
 import co.aikar.timings.MinecraftTimings; // Paper
+import com.empireminecraft.customevents.LivingEntityArmorProtectEvent; // EMC
 
 public abstract class EntityLiving extends Entity {
 
@@ -1576,7 +1577,12 @@ public abstract class EntityLiving extends Entity {
     public int getArmorStrength() {
         AttributeInstance attributeinstance = this.getAttributeInstance(GenericAttributes.ARMOR);
 
-        return MathHelper.floor(attributeinstance.getValue());
+        // EMC start
+        final double value = attributeinstance.getValue();
+        LivingEntityArmorProtectEvent event = new LivingEntityArmorProtectEvent(this.getBukkitLivingEntity(), value);
+        event.callEvent();
+        return MathHelper.floor(event.getArmorValue());
+        // EMC end
     }
 
     protected void damageArmor(float f) {}
